src/commands/dev.ts

```diff
@@ -370,23 +395,36 @@ export async function command(commandOptions: CommandOptions) {
         return;
       }
 
-      if (proxy) {
-        for (const workerConfig of config.scripts) {
-          if (workerConfig.type !== 'proxy') {
-            continue;
+      config.proxy.forEach(([pathPrefix, proxyOptions], i) => {
+        if (proxyOptions.wsListener) return;
+
+        (req.connection as any).server.on('upgrade', (req, socket, head) => {
+          if (
+            (proxyOptions.ws || proxyOptions.target.startsWith('ws')) &&
+            shouldProxy(pathPrefix, req)
+          ) {
+            try {
+              proxyOptions.proxy.ws(req, socket, head, proxyOptions);
```

Comment:
I think it makes more sense to send `proxyOptions` to the `createProxyServer()`  call, and then we don't need to worry about it here. From reading the README, that seems to be the better supported usecase.